### PR TITLE
Ability for Tabs tor respond to aria-disabled attr

### PR DESCRIPTION
**Pull request by ShadowedMists**
Adds the ability for tabs to respond appropriately to the disabled CSS class and disabled attribute.

### CODE REVIEW

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 2e548f77edd3d542ef499bd5691ede1e82141112
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/64112146

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by cvrebert** [comment]
I don't think an `<li>` can be `:disabled` ?

**Comment by cvrebert** [comment]
Also, this needs a unit test.

**Comment by mdo** [comment]
Indeed it cannot @cvrebert.

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 58ae01f6c3cf65c91aa5afe6e96839203e951429
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/64142429

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by cvrebert** [comment]
@ShadowedMists Please fix the trailing whitespace errors: https://travis-ci.org/twbs/bootstrap/jobs/64142381

**Comment by ShadowedMists** [comment]
@cvrebert whitespace has been removed as per code style tests.

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 4791878d260905e3bb4378fae3b388ab9f212a08
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/64143872

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by cvrebert** [comment]
CC: @fat for review

**Comment by patrickhlauke** [comment]
Got to admit, not a fan of this. The link in the disabled tab is still focusable with the keyboard, but gives no indication that it's disabled to, say, a screenreader user. At the very least, the link itself it should also have an `aria-disabled="true"` attribute.

**Comment by cvrebert** [comment]
This would need an example in the documentation, come to think of it.

**Comment by ShadowedMists** [comment]
The check can be based on the aria attribute if desired. Just a feature request to have some disabled tab functionality. Nav already supplies the CSS. I have written several variants of tabs that feature a "disabled" tab that the user may see, but the user is prohibited from clicking. I have written them because Tabs doesn't support the feature without a custom Bootstrap function. I would like to see the feature added in some capacity. This would also help applications where an onscreen wizard needs to see steps, but cannot yet proceed due to validation.  

If an example is needed in documentation, is that something that I should commit as well?

**Comment by cvrebert** [comment]
> If an example is needed in documentation, is that something that I should commit as well?

Yes please. Adding a disabled tab to the existing example would probably be sufficient: https://github.com/twbs/bootstrap/blob/master/docs/_includes/js/tabs.html

**Comment by cvrebert** [comment]
@patrickhlauke I guess we ought to add `aria-disabled="true"` to http://getbootstrap.com/components/#nav-disabled-links then?

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 28c60c535f3809ea5d25e282652a94faef54715f
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/64185559

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

File: docs/_includes/js/tabs.html [hunk 1]
```diff
@@ -28,6 +29,9 @@ <h2 id="tabs-examples">Example tabs</h2>
       <div role="tabpanel" class="tab-pane fade" id="dropdown2" aria-labelledBy="dropdown2-tab">
         <p>Trust fund seitan letterpress, keytar raw denim keffiyeh etsy art party before they sold out master cleanse gluten-free squid scenester freegan cosby sweater. Fanny pack portland seitan DIY, art party locavore wolf cliche high life echo park Austin. Cred vinyl keffiyeh DIY salvia PBR, banh mi before they sold out farm-to-table VHS viral locavore cosby sweater. Lomo wolf viral, mustache readymade thundercats keffiyeh craft beer marfa ethical. Wolf salvia freegan, sartorial keffiyeh echo park vegan.</p>
       </div>
+	  <div role="tabpanel" class="tab-pane fade" id="disabled" aria-labelledBy="disabled-tab">
```
Comment (cvrebert) [inline docs/_includes/js/tabs.html]: I think using `disabled` as the exact `id` might cause some confusion, given that there's also a `.disabled` class. Perhaps use something like `disabled-example` for the `id`?

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 19e744bc5757098c2c6aefdb252be186f8ac94b5
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/64187310

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by ShadowedMists** [comment]
@cvrebert @patrickhlauke  
About the Nav, would you not then need a whole JavaScript plugin just to prevent navigation on disabled links? Simply to do this?:

    $('.nav a').on('click', function (e) {
      if ($(this).attr('aria-disabled') === 'true') {
        e.preventDefault()
        e.stopPropagation()
        return false
      }
    })

**Comment by mdo** [comment]
[**Bootstrap 3 is no longer being officially developed or supported.**](https://github.com/twbs/bootstrap/issues/20631)

All work has moved onto our next major release, v4. As such, this issue or pull request is being closed as a "won't fix." For additional help and support, we recommend [utilizing our community resources](https://github.com/twbs/bootstrap/blob/master/README.md#community). Thanks for your understanding, and see you on the other side of v4!

<3,
@mdo and team

**Pull request closed without merging**